File: src/entity/voiture.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entity;

/**
 *
 * @author tekay
 */
public class voiture {
     private int matricule;
    private String etat;
    private int annes;
    private int prix_h;
    private int id_m;

    public voiture() {
    }

    public voiture(int matricule, String etat, int annes, int prix_h, int id_m) {
        this.matricule = matricule;
        this.etat = etat;
        this.annes = annes;
        this.prix_h = prix_h;
        this.id_m = id_m;
    }

    public voiture(String etat, int annes, int prix_h, int id_m) {
        this.etat = etat;
        this.annes = annes;
        this.prix_h = prix_h;
        this.id_m = id_m;
    }

    public int getMatricule() {
        return matricule;
    }

    public void setMatricule(int matricule) {
        this.matricule = matricule;
    }

    public String getEtat() {
        return etat;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    public int getAnnes() {
        return annes;
    }

    public void setAnnes(int annes) {
        this.annes = annes;
    }

    public int getPrix_h() {
        return prix_h;
    }

    public void setPrix_h(int prix_h) {
        this.prix_h = prix_h;
    }

    public int getId_m() {
        return id_m;
    }

    public void setId_m(int id_m) {
        this.id_m = id_m;
    }





   
}

File: src/Entity/Voiture.php
<?php

namespace App\Entity;

use App\Repository\VoitureRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
#[ORM\Entity(repositoryClass: VoitureRepository::class)]
class Voiture
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message:"etat is required")]
    private ?string $etat = null;

    #[ORM\Column]
    #[Assert\NotBlank(message:"annes is required")]
    private ?int $annes = null;

    #[ORM\Column]
    #[Assert\NotBlank(message:"prix is required")]
    private ?int $prix_h = null;

    #[ORM\ManyToOne(inversedBy: 'voiture')]
    private ?Modele $modele = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getEtat(): ?string
    {
        return $this->etat;
    }

    public function setEtat(string $etat): self
    {
        $this->etat = $etat;

        return $this;
    }

    public function getAnnes(): ?int
    {
        return $this->annes;
    }

    public function setAnnes(int $annes): self
    {
        $this->annes = $annes;

        return $this;
    }

    public function getPrixH(): ?int
    {
        return $this->prix_h;
    }

    public function setPrixH(int $prix_h): self
    {
        $this->prix_h = $prix_h;

        return $this;
    }

    public function getModele(): ?Modele
    {
        return $this->modele;
    }

    public function setModele(?Modele $modele): self
    {
        $this->modele = $modele;

        return $this;
    }
}
